Source/JavaScriptCore/dfg/DFGByteCodeParser.cpp

```diff
@@ -5791,6 +5809,21 @@ void ByteCodeParser::parseBlock(unsigned limit)
             for (unsigned i = 0; i < m_inlineStackTop->m_codeBlock->numVars(); ++i)
                 set(virtualRegisterForLocal(i), undefined, ImmediateNakedSet);
 
+            if (codeBlock->hasTailCalls()) {
+                m_inlineStackTop->m_entryBlockForRecursiveTailCall = allocateUntargetableBlock();
+                addToGraph(Jump, OpInfo(m_inlineStackTop->m_entryBlockForRecursiveTailCall));
+                m_currentBlock = m_inlineStackTop->m_entryBlockForRecursiveTailCall;
+            }
```

Comment:
Nit: Maybe add a comment to the commit message that you have to add this new block because you can't search for this bytecode lazily now.

Reply:
Good point, will do